### PR TITLE
Xun Serialization

### PR DESCRIPTION
**Pull request by JensGM**

### CODE REVIEW

File: xun/serialization/yaml.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    float,
+    datetime.date,
+    datetime.datetime,
+)
+
+
+def representer(functor):
+    def _representer(dumper, value):
+
+        # The inverse functor will be used for construction
```
Comment (JensGM) [inline xun/serialization/yaml.py:42]: Outdated comment

File: xun/tests/test_serialization.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    recursive_dict = {1: 1}
+    recursive_dict['self'] = recursive_dict
+    yml = xun.serialization.dumps(recursive_dict)
+    loaded = xun.serialization.loads(yml)
+
+    assert loaded[1] == 1
+    assert loaded['self'] is loaded
+
+
+def test_hash_is_order_independent():
```
Comment (JensGM) [inline xun/tests/test_serialization.py:20]: test_hash_is_equal_to_inverse_hash

File: xun/tests/test_serialization.py [hunk 3]
```diff
@@ -88,3 +89,35 @@ def test_callnode_serialization():
...
+    assert type(loaded) == type(mean)
+
+
+def test_pandas_series_datetime_index_serialization():
+    date_range = pd.date_range(datetime.now(), periods=10, tz='UTC')
+    index = pd.DatetimeIndex(date_range, freq=None)
+    series = pd.Series(np.random.randn(10), index=index)
+    yml = xun.serialization.dumps(series)
+    loaded = xun.serialization.loads(yml)
+    print(yml)
```
Comment (JensGM) [inline xun/tests/test_serialization.py]: print left over

File: xun/tests/test_serialization.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+
+def test_serialization():
+    recursive_dict = {1: 1}
+    recursive_dict['self'] = recursive_dict
+    yml = xun.serialization.dumps(recursive_dict)
+    loaded = xun.serialization.loads(yml)
+
+    assert loaded[1] == 1
+    assert loaded['self'] is loaded
```
Comment (JensGM) [inline xun/tests/test_serialization.py:17]:
```suggestion
    assert yml != recursive_dict
    assert loaded[1] == 1
    assert loaded['self'] is loaded
```

**Review by vero-so — APPROVED** [review]
Fix the leftover prints, but otherwise looks good and can be merged.

**Pull request merged**